public/api/v1/index.php

```diff
@@ -1002,6 +1006,61 @@ function echo_main_dashboard_JSON($project_instance, $date)
                 }
             }
 
+            if ($filter_on_labels) {
+                $label_filter_sql = get_filterdata_from_request('queryTests.php')['sql'];
+                $label_query_base = "SELECT build2test.status, test.name AS testname
+                                    FROM build AS b
+                                    JOIN build2test ON (b.id = build2test.buildid)
+                                    JOIN test ON (test.id = build2test.testid)
+                                    WHERE b.projectid = '" . $project_array['id'] . "' ";
+
+                $label_filter_query = $label_query_base . $parent_clause . $date_clause . ' ' .
+                                     $label_filter_sql . $limit_sql;
+
+                $labels_result = pdo_query($label_filter_query);
+                $nnotrun = 0;
+                $nfail = 0;
+                $npass = 0;
+                while ($label_row = pdo_fetch_array($labels_result)) {
+                    switch ($label_row['status']) {
+                        case 'passed':
+                            $npass++;
+                            break;
+                        case 'failed':
+                            $nfail++;
+                            break;
+                        case 'notrun':
+                            $nnotrun++;
+                            break;
+                    }
+                }
+
+                $only_new_tests = ' AND build2test.newstatus=1 ';
+                $label_filter_query = $label_query_base . $only_new_tests .
+                                    $parent_clause . $date_clause . ' ' .
+                                    $label_filter_sql . $limit_sql;
+                $labels_result = pdo_query($label_filter_query);
+                $test_response['nfaildiffp'] = 0;
+                $test_response['nfaildiffn'] = 0;
+                $test_response['npassdiffp'] = 0;
+                $test_response['npassdiffn'] = 0;
+                $test_response['nnotrundiffp'] = 0;
+                $test_response['nnotrundiffn'] = 0;
+                while ($label_row = pdo_fetch_array($labels_result)) {
+                    switch ($label_row['status']) {
+                        case 'passed':
+                            $test_response['npassdiffp']++;
+                            break;
+                        case 'failed':
+                            $test_response['nfaildiffp']++;
+                            break;
+                        case 'notrun':
+                            $test_response['nnotrundiffp']++;
+                            break;
+                    }
+                }
+            }
```

Comment:
The negative subscripts never get set here, but I'm okay with this (at least for now).  These numbers would be pretty difficult to compute based on how this data is currently stored.